app/core/services/user_service.py

```
@@ -1,9 +1,53 @@
-	user_info = UserInfo(**user_data.model_dump())
-	user = await user_repo.create_user(user_info)
-	return UserResponse.model_validate(user)
+	try:
+		user_info = UserInfo(**user_data.model_dump())
+		user_info.password = Hash.get_hashed_password(user_data.password)
+		user_info.roles = [UserRoles.USER]
+		user = await user_repo.create_user(user_info)
+		return UserResponse.model_validate(user)
+	except IntegrityError:
+		await user_repo.session.rollback()
+		raise UserAlreadyExistsError('A user with this username or email already exists.')
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)